### PR TITLE
Add namespace status conditions

### PR DESCRIPTION
**Pull request by wozniakjan**
a docs update following the implementation in https://github.com/kubernetes/kubernetes/pull/73405

### CODE REVIEW

File: contributors/design-proposals/architecture/namespaces.md [hunk 1]
```diff
@@ -191,6 +191,52 @@ If the *namespace controller* sees a *Namespace* whose
...
+resource deletion, remembers the errors that occurred and report back via
+**namespace status condition**. Some errors can be transient and will
+auto-resolve in the following cleanup runs, others may require manual
+intervention.
+
+There are following status conditions reporting on the process of namespace
+termination:
+- `NamespaceDeletionDiscoveryFailure` reports on errors during the first phase
+   of namespace termination - 
+   [resource discovery](../api-machinery/api-group.md).
```
Comment (wozniakjan) [inline contributors/design-proposals/architecture/namespaces.md]: I am not entirely satisfied with the link selection here but I couldn't find better page explaining the resource discovery. And it feels like this should be hyperlinked

File: contributors/design-proposals/architecture/namespaces.md [hunk 2]
```diff
@@ -191,6 +191,52 @@ If the *namespace controller* sees a *Namespace* whose
 list is empty, it will signal the server to permanently remove the *Namespace*
 from storage by sending a final DELETE action to the API server.
 
+There are situations where the *namespace controller* is unable to guarantee 
+cleanup of all resources. During a cleanup run, it attempts a best-effort 
+resource deletion, remembers the errors that occurred and report back via
```
Comment (geekygirldawn) [inline contributors/design-proposals/architecture/namespaces.md]: probably should be "reports back"?

File: contributors/design-proposals/architecture/namespaces.md [hunk 3]
```diff
@@ -191,6 +191,52 @@ If the *namespace controller* sees a *Namespace* whose
...
+**namespace status condition**. Some errors can be transient and will
+auto-resolve in the following cleanup runs, others may require manual
+intervention.
+
+There are following status conditions reporting on the process of namespace
```
Comment (geekygirldawn) [inline contributors/design-proposals/architecture/namespaces.md]: This wording seems a little awkward - "There are following status conditions ...". Maybe it should be "There are the following status conditions ..." or "These are the status conditions ..."

File: contributors/design-proposals/architecture/namespaces.md [hunk 4]
```diff
@@ -191,6 +191,52 @@ If the *namespace controller* sees a *Namespace* whose
...
+termination:
+- `NamespaceDeletionDiscoveryFailure` reports on errors during the first phase
+   of namespace termination - 
+   [resource discovery](../api-machinery/api-group.md).
+- `NamespaceDeletionGroupVersionParsingFailure` reports on errors that happen
+   when parsing the [GVK](../api-machinery/api-group.md) 
+   of all discovered resources.
+- `NamespaceDeletionContentFailure` reports on errors preventing the controller
+   to delete resources belonging to successfully discovered and parsed GVK.
```
Comment (geekygirldawn) [inline contributors/design-proposals/architecture/namespaces.md]: "preventing the controller to delete resources belonging" might be better as "preventing the controller from deleting resources belonging

File: contributors/design-proposals/architecture/namespaces.md (hunk 3 above)
Comment (wozniakjan) [reply contributors/design-proposals/architecture/namespaces.md]: I like the second wording you have mentioned very much - changing to `These are the status conditions ...`

**Comment by wozniakjan** [comment]
thank you very much @geekygirldawn, I think your suggestions were spot on. Greatly appreciate your input.

**Comment by jdumars** [comment]
Wondering if this is adequate to proceed with, and improve over time.

**Comment by lavalamp** [comment]
/lgtm

@deads2k added or is adding additional ones IIRC.

**Comment by gkarthiks** [comment]
hi @wozniakjan Since we have a KEP process, can we please merge this soon or close this and create a KEP?

**Comment by wozniakjan** [comment]
@gkarthiks I thought KEP (kubernetes enhancement proposal) is for new functionality? Namespace status conditions have been merged and released in 1.16. This PR is just to update the docs reflect the implementation.

**Comment by gkarthiks** [comment]
@wozniakjan gotcha, I was cleaning up all the PRs saying `kind/design`.

**Comment by gkarthiks** [comment]
in this case i  will update the label to documentation.

/kind documentation

**Comment by gkarthiks** [comment]
hi @geekygirldawn / @jdumars can you please review and merge this MR. As we are cleaning the PRs that are labeled with `kind/design` and suggesting to move them over to KEP process.

**Comment by jdumars** [comment]
/approve
/hold

Feel free to remove the hold as you see fit

**Comment by wozniakjan** [comment]
/unhold

> Feel free to remove the hold as you see fit

if this wasn't aimed at me, ping here or on slack if you'd like me to do any followup action :)

**Pull request merged**